### PR TITLE
Fixes #10878: let Number Notation be insensitive to the implicit arguments of its parameters

### LINKED ISSUE #10878: Numeral Notation is sensitive to the implicit status of `option`
#### Description of the problem

```coq
Require Import Coq.ZArith.ZArith.

Definition Zto_pos_opt (v : Z) : option positive
  := match v with
     | Zpos v => Some v
     | _ => None
     end.
Declare Scope mypos_scope.
Declare Scope mypos_scope2.
Numeral Notation positive Zto_pos_opt Zpos : mypos_scope. (* success *)
Arguments option {_}.
Fail Numeral Notation positive Zto_pos_opt Zpos : mypos_scope2.
(* Error:
Zto_pos_opt should go from Decimal.int to positive or (option positive).
Instead of Decimal.int, the types Decimal.uint or Z or Int63.int or Decimal.decimal could be used (you may need to require BinNums or Decimal or Int63 first).
 *)
```
This is due to the use of `mkRefC`:
https://github.com/coq/coq/blob/ac862fb5ae8eb15c15f81817d78ba8db4430ea8b/plugins/syntax/numeral.ml#L117

#### Coq Version

master

### PR DESCRIPTION
**Pull request by herbelin**
Fixes / closes #10878. See discussion there.

- [x] Added / updated **test-suite**.
- [ ] Added **changelog**.

### CODE REVIEW

File: dev/ci/ci-vst.sh [hunk 1]
```diff
@@ -11,9 +11,9 @@ if [ "$DOWNLOAD_ONLY" ]; then exit 0; fi
...
 
-ulimit -s
-ulimit -s 65536
-ulimit -s
+sudo ulimit -s
+sudo ulimit -s 65536
+sudo ulimit -s
 ( cd "${CI_BUILD_DIR}/vst"
   make IGNORECOQVERSION=true
 )
```
Comment (Zimmi48) [inline dev/ci/ci-vst.sh:19]: These changes should not be part of this PR. Move them to another PR if you want them to be discussed.

Comment (herbelin) [reply dev/ci/ci-vst.sh]: The current PR uselessly includes PR #17902 which itself mistakenly included this debugging code. Now fixed.

**Review by proux01 — APPROVED** [review]

**Comment by proux01** [comment]
@coqbot run full ci

**Comment by JasonGross** [comment]
@coqbot bench

**Comment by proux01** [comment]
> @coqbot bench

@JasonGross what kind of performance issue do you expect?

**Comment by proux01** [comment]
CI green and the bench show nohing (I still don't know what we were expecting there) so

@coqbot merge now

**Comment by JasonGross** [comment]
> @JasonGross what kind of performance issue do you expect?

I think I was confusing implicits in inductive types with implicits in constructors?  I don't quite recall what I was thinking.

Also, wasn't there some trick for number notations with inductive families like `Fin.t` that relied on making the argument implicit and maximally inserted?  Or am I misremembering?

**Comment by proux01** [comment]
AFAIR, those were the implicit arguments of the "constructors", not the type itself.

**Pull request merged**